src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -341,9 +339,9 @@ private async Task ProcessHeadersFrame(FrameHeader frameHeader)
 
             _hpackDecoder.CompleteDecode();
 
-            http2Stream.OnResponseHeadersComplete(endStream);
+            http2Stream?.OnResponseHeadersComplete(endStream);
 
-            if (endStream)
+            if (endStream && http2Stream != null)
             {
                 RemoveStream(http2Stream);
             }
```

Comment:
Nit: just to save a null check (and maybe make the logic a little more clear), maybe instead make this:
```C#
if (http2Stream != null)
{
    httpStream.OnResponseHeadersComplete(endStream);
    if (endStream)
    {
        RemoveStream(http2Stream);
    }
}
```
?